### DISCUSSION TITLE
How to declare H3Event or CompatibilityEvent type in middleware/myauth.ts

### DISCUSSION THREAD

**Discussion by xiangnanscu** [nuxt/nuxt#16738] [General]
Hello, I'm trying write a auth middleware `middleware/myauth.ts` like this:
```ts
const login = (event:H3Event, user: object) => {
  setCookie()
}
// or

const login = (event:CompatibilityEvent, user: object) => {
  setCookie()
}
```
So I can use this `login` in server/api route like this 
```ts
export default defineEventHandler(async (event) => {
  login(event, {id: 1, username:'foo'})
})
```
But the IDE shows `Cannot find name 'H3Event'`:  
![image](https://user-images.githubusercontent.com/8246344/192670533-d8ddf18b-d5aa-4abb-98d6-6b8a92fecb03.png)

**Comment by ihkawiss** [comment]
I faced the same question, this worked for me:
`import type { H3Event } from 'h3'`

**Comment by FrankFang** [comment]
It's really strange that the documentation doesn't mention this.
